rotkehlchen/chain/optimism/modules/velodrome/decoder.py

```diff
@@ -36,11 +64,224 @@ def __init__(
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.counterparty = CPT_VELODROME
+                event.notes = f'Deposit {event.balance.amount} {crypto_asset.symbol} in velodrome pool {event.address}'  # noqa: E501
+                event.product = EvmProduct.POOL
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.address == ZERO_ADDRESS
+            ):
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.counterparty = CPT_VELODROME
+                event.notes = f'Receive {event.balance.amount} {crypto_asset.symbol} after depositing in velodrome pool {tx_log.address}'  # noqa: E501
+                event.product = EvmProduct.POOL
+            elif (
+                event.event_type == HistoryEventType.INFORMATIONAL and
+                event.event_subtype == HistoryEventSubType.APPROVE
+            ):
+                event.counterparty = CPT_VELODROME
+
+        return DEFAULT_DECODING_OUTPUT
 
-    def _decode_velodrome_gauge_events(self) -> None:
-        pass
+    def _decode_remove_liquidity_events(
+            self,
+            tx_log: 'EvmTxReceiptLog',
+            decoded_events: list['EvmEvent'],
+    ) -> DecodingOutput:
+        """Decodes events that remove liquidity from a velodrome pool"""
+        for event in decoded_events:
+            try:
+                crypto_asset = event.asset.resolve_to_crypto_asset()
+            except (UnknownAsset, WrongAssetType):
+                self.notify_user(event=event, counterparty=CPT_VELODROME)
+                raise
+
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.address in self.pools
+            ):
+                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                event.counterparty = CPT_VELODROME
+                event.notes = f'Return {event.balance.amount} {crypto_asset.symbol}'
+                event.product = EvmProduct.POOL
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.address in self.pools
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.counterparty = CPT_VELODROME
+                event.notes = f'Remove {event.balance.amount} {crypto_asset.symbol} from velodrome pool {tx_log.address}'  # noqa: E501
+                event.product = EvmProduct.POOL
+            elif (
+                event.event_type == HistoryEventType.INFORMATIONAL and
+                event.event_subtype == HistoryEventSubType.APPROVE
+            ):
```

Comment:
the information events are also emitted by the pool address? If that is this case you can do the check `event.address in self.pools` (it might be in the topic) earlier and exit faster

Reply:
No they are emitted from the router.